packages/lab/src/date-input/DateInput.tsx

```diff
@@ -0,0 +1,219 @@
+import { clsx } from "clsx";
+import {
+  ChangeEvent,
+  ComponentPropsWithoutRef,
+  FocusEvent,
+  forwardRef,
+  InputHTMLAttributes,
+  Ref,
+  useState,
+} from "react";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+
+import dateInputCss from "./DateInput.css";
+import { makePrefixer, useControlled, useFormFieldProps } from "@salt-ds/core";
+
+const withBaseName = makePrefixer("saltDateInput");
+
+export interface DateInputProps
+  extends Omit<ComponentPropsWithoutRef<"div">, "defaultValue">,
+    Pick<
+      ComponentPropsWithoutRef<"input">,
+      "disabled" | "value" | "defaultValue" | "placeholder"
+    > {
+  /**
+   * The marker to use in an empty read only DateInput.
+   * Use `''` to disable this feature. Defaults to '—'.
+   */
+  emptyReadOnlyMarker?: string;
+  /**
+   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dateInput#Attributes) applied to the `input` elements.
+   */
+  inputProps?: InputHTMLAttributes<HTMLInputElement>;
+  /**
+   * Optional ref for the dateInput component
+   */
+  inputRef?: Ref<HTMLInputElement>;
+  /**
+   * If `true`, the component is read only.
+   */
+  readOnly?: boolean;
+  /**
+   * Alignment of text within container. Defaults to "left"
+   */
+  textAlign?: "left" | "center" | "right";
+  /**
+   * Validation status.
+   */
+  validationStatus?: "error" | "warning" | "success";
+  /**
+   * Styling variant. Defaults to "primary".
+   */
+  variant?: "primary" | "secondary";
+  /**
+   * Parser to format the input as date.
+   */
+  dateParser?: (input: string) => string;
```

Comment:
@joshwooding do you think this might be a problem with cal? guess it could be date -> string and we always do the date part within the component...